### PR TITLE
GH-43160: [Swift] Add Struct Array

### PR DESCRIPTION
**Pull request by abandy**
### Rationale for this change
Adding StructArray for struct type impl

### What changes are included in this PR?
Added StructArray impl

* GitHub Issue: #43160

### CODE REVIEW

File: swift/Arrow/Sources/Arrow/ArrowArray.swift [hunk 1]
```diff
@@ -14,26 +14,25 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
-
```
Comment (kou) [inline swift/Arrow/Sources/Arrow/ArrowArray.swift:17]: Is it intentional change?

File: swift/Arrow/Sources/Arrow/ArrowArray.swift [hunk 2]
```diff
@@ -233,3 +273,58 @@ public class BinaryArray: ArrowArray<Data> {
...
     }
 }
+
+public class StructArray: ArrowArray<[Any?]> {
+    public private(set) var arrowFields: [ArrowArrayHolder]?
+    public required init(_ arrowData: ArrowData) {
+        super.init(arrowData)
+    }
+
+    public func initialize() throws -> StructArray {
```
Comment (kou) [inline swift/Arrow/Sources/Arrow/ArrowArray.swift:283]: I'm not familiar with Swift but why do we need to define `initialize()`?
Can we do this in `init()`? (`init()` should not throw?)

**Review by kou — COMMENTED** [review]
Do we need to add new  tests for `StructArray`?

File: swift/Arrow/Sources/Arrow/ArrowArray.swift (hunk 2 above)
Comment (abandy) [reply swift/Arrow/Sources/Arrow/ArrowArray.swift:283]: Correct, ArrowArray's init does not throw.  The ArrowArrayHolderImpl.loadArray method can throw so in order to allow this to be thrown I added the initialize method.  (I looked at adding a convenience init method but the throws keyword isn't considered in overloading.)

File: swift/Arrow/Sources/Arrow/ArrowArray.swift (hunk 1 above)
Comment (abandy) [reply swift/Arrow/Sources/Arrow/ArrowArray.swift:17]: It is not, I will add the space back in.

**Comment by abandy** [comment]
> Do we need to add new tests for `StructArray`?

I do have test but they require the builder which will be included in the next PR.

**Review by kou — APPROVED** [review]
+1

> I do have test but they require the builder which will be included in the next PR.

It makes sense.

**Pull request merged**